cargo-shuttle/src/lib.rs

```diff
@@ -82,42 +82,17 @@ impl Shuttle {
 
     async fn init(&self, args: InitArgs) -> Result<()> {
         // Interface with cargo to initialize new lib package for shuttle
-        let opts = NewOptions::new(None, false, true, args.path.clone(), None, None, None)?;
-        let cargo_config = cargo::util::config::Config::default()?;
-        let init_result = cargo::ops::init(&opts, &cargo_config)?;
-        // Mimick `cargo init` behavior and log status or error to shell
-        cargo_config
-            .shell()
-            .status("Created", format!("{} (shuttle) package", init_result))?;
-
-        // Read Cargo.toml into a `Document`
-        let cargo_path = args.path.join("Cargo.toml");
-        let mut cargo_doc = read_to_string(cargo_path.clone())?.parse::<Document>()?;
-
-        // Remove empty dependencies table to re-insert after the lib table is inserted
-        cargo_doc.remove("dependencies");
-
-        // Create an empty `[lib]` table
-        cargo_doc["lib"] = Item::Table(Table::new());
-
-        // Fetch the latest shuttle-service version from crates.io
-        let manifest_path = find(Some(&args.path)).unwrap();
-        let url = registry_url(manifest_path.as_path(), None).expect("Could not find registry URL");
-        let latest_shuttle_service =
-            get_latest_dependency("shuttle-service", false, &manifest_path, Some(&url))
-                .expect("Could not query the latest version of shuttle-service");
-        let shuttle_version = latest_shuttle_service
-            .version()
-            .expect("No latest shuttle-service version available");
-
-        // Insert shuttle-service to `[dependencies]` table
-        let mut dep_table = Table::new();
-        dep_table["shuttle-service"]["version"] = value(shuttle_version);
-        cargo_doc["dependencies"] = Item::Table(dep_table);
-
-        // Truncate Cargo.toml and write the updated `Document` to it
-        let mut cargo_toml = File::create(cargo_path)?;
-        cargo_toml.write_all(cargo_doc.to_string().as_bytes())?;
+        let path = args.path.clone();
+        init::cargo_init(path.clone())?;
+        init::process_cargo_init(path.clone())?;
+
+        let framework = init::get_framework(&args);
+        match framework {
+            Some(framework) => {
+                init::framework_init(&path, framework)?;
+            },
+            None => {},
```

Comment:
Is the idea here to do a no-op when the user did not choose a framework?

Reply:
Yes, no-op if the framework isn't there since `cargo init --lib` was called and that suffices. Any suggestion to improve?

Reply:
Nope, this fine. Just wanted to make sure it is intentional :smile: